minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/DockerBuildTask.java

```diff
@@ -158,29 +184,47 @@ public void execDockerBuild() throws IOException, InterruptedException {
    * Builds the default tag for the built image in the form: {@code
    * ${project.group}/${project.name}:${project.version}}.
    *
-   * <p>If {@code ${project.name}}
-   *
    * <p>If {@code ${project.group}} is empty, {@code ${project.group}/} will not be included.
    *
    * <p>If {@code ${project.version}} is empty, {@code :${project.version}} will not be included.
    *
-   * <p>Replaces all instances of ':' in group, name, or version with '.'.
-   *
-   * @return the built tag
-   * @throws GradleException
+   * @return the built tag, or an empty string if any part does not satisfy Docker tag rules
+   * @see <a href="https://docs.docker.com/engine/reference/commandline/tag/#description">docker
+   *     tag</a>
    */
   // @VisibleForTesting
   String buildDefaultTag() {
-    String group = getProject().getGroup().toString().replace(':', '.');
-    String projectName = getProject().getName().toString().replace(':', '.');
-    String projectVersion = getProject().getVersion().toString().replace(':', '.');
+    String group = getProject().getGroup().toString();
+    String projectName = getProject().getName().toString();
+    String projectVersion = getProject().getVersion().toString();
 
     StringBuilder tagBuilder = new StringBuilder();
+
     if (!group.isEmpty()) {
+      // Checks if project.group can be used as part of the image name.
+      if (!group.matches(NAME_COMPONENT_REGEX)) {
+        logger.warn(
+                "Default image tag could not be generated because project.group is not a valid name component");
+        return "";
+      }
       tagBuilder.append(group).append("/");
     }
+
+    // Checks if project.name can be used as part of the image name.
+    if (!projectName.matches(NAME_COMPONENT_REGEX)) {
+      logger.warn(
+              "Default image tag could not be generated because project.name is not a valid name component");
+      return "";
+    }
     tagBuilder.append(projectName);
+
     if (!"unspecified".equals(projectVersion)) {
+      // Checks if project.version can be used as the image tag name.
+      if (!projectVersion.matches(TAG_NAME_REGEX)) {
+        logger.warn(
+                "Default image tag could not be generated because project.version is not a valid tag name");
+        return "";
```

Comment:
Wondering if we should return `null` instead of `""` so that the build fails if they don't override it.
Fails because `@Input` on the `DockerBuildTask` wont have a value.

It might look like
```
./gradlew minikubeDockerBuild
WARNING: "blah blah blah, bad tag name"
Task Failed : minikubeDockerBuild because 'tag' was not defined.
```

Reply:
Yea that sounds good. Then, if the user truly wanted to not automatically tag the image, they can just set tag to "" explicitly.